### PR TITLE
#20108 remove "create partitions" button for MySQL/MariaDB

### LINKED ISSUE #20108: [MySQL/MariaDB] do not show empty default partitions
According to the [documentation](https://dev.mysql.com/doc/refman/5.7/en/information-schema-partitions-table.html), any table without created partitions has a default partition:

> A table using any storage engine other than [NDB](https://dev.mysql.com/doc/refman/5.7/en/mysql-cluster.html) and which is not partitioned has one row in the [PARTITIONS](https://dev.mysql.com/doc/refman/5.7/en/information-schema-partitions-table.html) table. However, the values of the PARTITION_NAME, SUBPARTITION_NAME, PARTITION_ORDINAL_POSITION, SUBPARTITION_ORDINAL_POSITION, PARTITION_METHOD, SUBPARTITION_METHOD, PARTITION_EXPRESSION, SUBPARTITION_EXPRESSION, and PARTITION_DESCRIPTION columns are all NULL. Also, the PARTITION_COMMENT column in this case is blank.

2023-05-26 12_22_52-DBeaver 23 0 3 - account

But it is confusing - to have empty partitions in all unpartitioned tables. Let's hide them.

### PR DESCRIPTION
**Pull request by LonwoLonwo**
Please check different places:

- [ ] In the table properties in the Partitions tab
- [ ] On the table in the Database navigator tree

### CODE REVIEW

**Review by E1izabeth — APPROVED** [review]

File: plugins/org.jkiss.dbeaver.ext.mysql/plugin.xml [hunk 1]
```diff
@@ -71,9 +71,9 @@
                                         </items>
                                     </items>
                                 </folder>
-                                <folder type="org.jkiss.dbeaver.ext.mysql.model.MySQLPartition" label="%tree.partitions.node.name" icon="#partitions" description="Table partitions" visibleIf="object.dataSource.supportsPartitions()">
+                                <folder type="org.jkiss.dbeaver.ext.mysql.model.MySQLPartition" label="%tree.partitions.node.name" icon="#partitions" description="Table partitions" visibleIf="object.dataSource.supportsPartitions()" virtual="true">
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.ext.mysql/plugin.xml]: Virtual nodes are supposed to be used for duplicated objects in a tree. How this helps in this case?

File: plugins/org.jkiss.dbeaver.ext.mysql/src/org/jkiss/dbeaver/ext/mysql/edit/MySQLPartitionTableManager.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver.ext.mysql.edit;
+
+public class MySQLPartitionTableManager extends MySQLTableManager {
+
+    @Override
+    public boolean canCreateObject(Object container) {
+        // Creating partitions separately is not supported in MySQL/MariaDB
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.ext.mysql/src/org/jkiss/dbeaver/ext/mysql/edit/MySQLPartitionTableManager.java:25]: I altho thought that we want to hide dummy partitioj in MySQL tables. Is it doable?

**Review by serge-rider — CHANGES_REQUESTED** [review]

File: plugins/org.jkiss.dbeaver.ext.mysql/src/org/jkiss/dbeaver/ext/mysql/edit/MySQLPartitionTableManager.java (hunk 2 above)
Comment (LonwoLonwo) [reply plugins/org.jkiss.dbeaver.ext.mysql/src/org/jkiss/dbeaver/ext/mysql/edit/MySQLPartitionTableManager.java:25]: Implemented.

**Comment by uslss** [comment]
verified

**Review by serge-rider — APPROVED** [review]

**Pull request merged**